java/src/main/java/com/google/appengine/tools/mapreduce/impl/shardedjob/ShardedJobRunner.java

```diff
@@ -49,20 +51,29 @@
  * @author ohler@google.com (Christian Ohler)
  *
  */
-@AllArgsConstructor(onConstructor_ = @Inject)
 @Log
 public class ShardedJobRunner implements ShardedJobHandler {
 
   static final int TASK_LOOKUP_BATCH_SIZE = 20;
 
-  static final long CONTROLLER_TASK_DELAY = Duration.ofSeconds(2).toMillis();
-  static final long WORKER_TASK_DELAY = Duration.ofSeconds(2).toMillis();
-
   @Getter
   final Provider<PipelineService> pipelineServiceProvider;
   @Getter
   final Datastore datastore;
 
+  @Inject
+  public ShardedJobRunner(Provider<PipelineService> pipelineServiceProvider, Datastore datastore) {
+    this.pipelineServiceProvider = pipelineServiceProvider;
+    this.datastore = datastore;
+  }
+
+  @Getter @Setter
+  private long controllerTaskDelay = Duration.ofSeconds(2).toMillis();
+  @Getter @Setter
+  private long workerTaskDelay = Duration.ofSeconds(2).toMillis();
+  @Getter @Setter
+  private long lockCheckTaskDelay = Duration.ofSeconds(60).toMillis();
```

Comment:
perhaps clear to leave as `Duration`, explicitly convert to millis if/when needed.